neuraxle/checkpoints.py

```diff
@@ -160,13 +167,15 @@ def set_checkpoint_path(self, path):
         if not os.path.exists(self.checkpoint_path):
             os.makedirs(self.checkpoint_path)
 
-    def should_resume(self, data_container: DataContainer) -> bool:
+    def should_resume(self, data_container: DataContainer, context: ExecutionContext) -> bool:
         """
         Whether or not we should resume the pipeline (if the checkpoint exists)
 
+        :param context: execution context
         :param data_container: data to resume
         :return:
         """
+        self.set_checkpoint_path(context.get_path())
```

Comment:
Could we avoid setting this path, and always getting it on-the-fly from the context to avoid any possible side effects and have more pure class methods?

Reply:
it is in the should resume method, se we can say it is already calculated on the fly. closing this